src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -698,8 +773,7 @@ protected CompletableFuture<ExecutionResult> completeValueForObject(ExecutionCon
         );
 
         // Calling this from the executionContext to ensure we shift back from mutation strategy to the query strategy.
-
-        return executionContext.getQueryStrategy().execute(executionContext, newParameters);
+        return executionContext.getQueryStrategy().executePolymorphic(executionContext, newParameters);
```

Comment:
This means that a materialised object can be completed via recursively calling back to the ES but via a special method that allows CFs or materialized values